src/api/AuthApi.test.ts

```diff
@@ -0,0 +1,58 @@
+import sinon, { SinonFakeXMLHttpRequest, SinonFakeXMLHttpRequestStatic } from 'sinon';
+import { expect } from 'chai';
+import { AuthApi } from './AuthApi';
+
+describe('AuthApi', () => {
+    let xhr: SinonFakeXMLHttpRequestStatic;
+    let instance: AuthApi;
+    const requests: SinonFakeXMLHttpRequest[] = [];
+
+    beforeEach(() => {
+        xhr = sinon.useFakeXMLHttpRequest();
+
+        // @ts-expect-error
+        global.XMLHttpRequest = xhr;
+
+        xhr.onCreate = (request: SinonFakeXMLHttpRequest) => {
+            requests.push(request);
+        };
+
+        instance = new AuthApi();
+    });
+
+    afterEach(() => {
+        requests.length = 0;
```

Comment:
лучше сбрасывать значение расшаренной переменной присваивая новое значение, не меняя его свойства.
```suggestion
    requests = []
```

1) так проще читается
2) соблюдается принцип иммутабельности данных
3) нужно будет изменить const на let на 8 строке